### ISSUE TITLE
Clear environment variables after changing context

### ISSUE THREAD

**Issue by shirk3y** [grn/bash-ctx#4]
Let's say that some context sets `PATH` variable, and when we switch to another context we shouldn't have it modified.
My idea is to store all variables before first context switch to a file, and after consecutive switches we just clear environment, source this stored file and then run context script.

How does it sound?

**Comment by shirk3y** [comment]
Maybe this can be solution: http://stackoverflow.com/questions/15222328/bash-reset-reloading-bash-completely-alias-and-function

**Comment by gregnavis** [comment]
A pre-release had such feature. However I decided to get rid of it and use `enter` and `leave` scripts instead. Do you think that this can be solved by:

```
# enter
OLD_PATH="$PATH"
export PATH="/my/bin:$PATH"

# leave
export PATH="${OLD_PATH}"
unset OLD_PATH
```

**Comment by gregnavis** [comment]
I must admit that I _do not_ like to save some environment variables. It introduces unnecessary variables (`OLD_PATH`), requires editing of `enter` and `leave` and is error-prone.

I think that I like your idea of storing the whole of environment. I used `printenv` for this purpose in the pre-release version. There are at least three things that we must take into consideration:
1. an existing variable is modified
2. an existing variable is unset
3. a new variable is defined

I think that variables from 1 and 2 can be restored (I'm not sure whether that's always the right thing to do though).

Should we unset new variables (point 3 above) upon leaving a context? This makes sense for variables like `RAILS_ENV`, e.g.:

```
$ ctx enter my-rails-app
$ export RAILS_ENV=staging
$ do_some_work
$ ctx leave
```

This would unset `RAILS_ENV` and I think this is desired in the case above. Is this _always_ desired?

**Comment by gregnavis** [comment]
Addressed in 1bf77e1.
